### DISCUSSION TITLE
Debugging corrupt data when trying to hydrate error

### DISCUSSION THREAD

**Discussion by Braunson** [livewire/livewire#2824] [1. Help]
**What seems to be the problem:**
I'm seeing this in production randomly using the latest Livewire

> Livewire\Exceptions\CorruptComponentPayloadException
> 
> Livewire encountered corrupt data when trying to hydrate the [auth.login] component. Ensure that the [name, id, data] of the Livewire component wasn't tampered with between requests.

**Steps to Reproduce:**
- Unable to consistently reproduce, happens randomly in production every so often.

**Are you using the latest version of Livewire:** Yes

**Do you have any screenshots or code examples:**
```php
<?php

namespace App\Http\Livewire\Auth;

use Livewire\Component;
use Illuminate\Support\Facades\Auth;

class Login extends Component
{
    /** @var string */
    public $email = '';

    /** @var string */
    public $password = '';

    /** @var bool */
    public $remember = false;

    public $returnTo = null;

    protected $queryString = ['returnTo'];

    public function authenticate()
    {
        $credentials = $this->validate([
            'email'    => ['required', 'email'],
            'password' => ['required'],
        ]);

        // Check for is active
        $details = array_merge($credentials, ['is_active' => 1]);

        if (! Auth::attempt($details, $this->remember)) {
            $this->addError('email', trans('auth.failed'));

            return;
        }

        // Grab the default for the user type
        $default = (new \App\Http\Controllers\Auth\LoginController)->redirectTo(false);

        // Helper to return to a specific page on request
        if (isset($this->redirectTo) && ! is_null($this->returnTo)) {
            $default = url($this->returnTo);
        }

        redirect()->intended($default);
    }

    public function render()
    {
        return view('livewire.auth.login')
            ->extends('theme::layouts.auth')
            ->section('content');
    }
}
```

I mentioned this in the Livewire Discord, and it was mentioned to log the `$fingerprint` and `$memoSansChildren` from the `ComponentChecksumManager.php` and I was going to extend/override the class to log this as Sentry context, but it was mentioned I could somehow capture this in a middleware. 

Ideally I didn't want to extend/override the core Livewire class, any suggestions on how to debug this as it appears randomly on production for no absolute reason, at least from what I can tell.

**Comment by calebporzio** [comment]
I agree there should be a better way to debug this error. It is a very tough problem however because of the nature of checksums. Livewire has no way of telling what part of the payload didn't match the checksum, only that it didn't match.

The only solution I can really think of is keeping track of the data that originally generated the checksum to later compare it against, but that would require overhead somewhere, so we would probably need to make that only happen in dev envs.

**Reply by Braunson** [reply]
Perhaps this could be an option (keeping track of the original data) on a per-component or `.env` flag basis? I'm unable to re-create it locally and it happens very randomly on production. The audience in my case isn't too technologically knowledgeable so I doubt they are modifying anything. The code is running on Heroku and has multiple "dynos", not sure if this could be related to the cause or not but all sessions are stored on the same Redis server, everything is on the same domain.

**Reply by calebporzio** [reply]
Is it's ovccurance correlated to deployments? I could see a deployment causing an issue for someone who hasn't fully refreshed their page

**Reply by Braunson** [reply]
No I can confirm it's not related to deployments. In the unlikely chance it is, they would have to have opened the page and waited ~1 week before submitting it. Even wouldn't there have to have been a change in Livewire assets within the deployed commits?

**Reply by calebporzio** [reply]
Gotcha, yeah I think you're right
